src/worker/jobs/downloads/clean_log_dedupe_table.rs

```diff
@@ -0,0 +1,109 @@
+use crate::schema::processed_log_files;
+use crate::worker::Environment;
+use anyhow::anyhow;
+use crates_io_worker::BackgroundJob;
+use diesel::prelude::*;
+use std::sync::Arc;
+
+/// This job is responsible for cleaning up old entries in the
+/// `processed_log_files` table.
+///
+/// Rows older than one week will be deleted.
+#[derive(Serialize, Deserialize)]
+pub struct CleanLogDedupeTable;
+
+impl BackgroundJob for CleanLogDedupeTable {
+    const JOB_NAME: &'static str = "clean_log_dedupe_table";
+    const QUEUE: &'static str = "downloads";
+
+    type Context = Arc<Environment>;
+
+    async fn run(&self, env: Self::Context) -> anyhow::Result<()> {
+        let conn = env.deadpool.get().await?;
+        conn.interact(run)
+            .await
+            .map_err(|err| anyhow!(err.to_string()))??;
+
+        Ok(())
+    }
+}
+
+fn run(conn: &mut PgConnection) -> QueryResult<()> {
+    let cut_off_date = chrono::Utc::now() - chrono::Duration::try_weeks(1).unwrap();
+
+    diesel::delete(processed_log_files::table.filter(processed_log_files::time.lt(cut_off_date)))
+        .execute(conn)?;
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::test_util::test_db_connection;
+    use chrono::{DateTime, Utc};
+    use insta::assert_debug_snapshot;
+
+    #[test]
+    fn test_cleanup() {
+        let (_test_db, conn) = &mut test_db_connection();
+
+        let now = chrono::Utc::now();
+        let hours = chrono::Duration::try_hours(1).unwrap();
+
```

Comment:
```suggestion
        let weeks = chrono::Duration::try_weeks(1).unwrap();

        insert(
            conn,
            vec![
                ("very-old-file", now - weeks * 4),
                ("old-file", now - weeks - hours),
                ("newish-file", now - weeks + hours),
                ("brand-new-file", now),
                ("future-file", now + weeks),
            ],
        );
```
I would suggest performing the arithmetic based on our cutoff value. This would be clearer and easier to follow.